src/os/bluestore/BlueStore.cc

```diff
@@ -4316,7 +4316,7 @@ int BlueStore::_open_bdev(bool create)
   assert(bdev == NULL);
   string p = path + "/block";
   bdev = BlockDevice::create(cct, p, aio_cb, static_cast<void*>(this), discard_cb, static_cast<void*>(this));
-  int r = bdev->open(p);
+  int r = bdev->open(p, create);
```

Comment:
I think it would be easier/cleaner to just add an if (discard_on_create) bdev->discard(...) after the three open() calls in _open_bdev().  A "create" mode on block device open feels like an awkward interface to me.